File: src/main/java/net/siegemc/core/items/items/TestHelmet.java
package net.siegemc.core.items.items;

import net.siegemc.core.Core;
import net.siegemc.core.items.ArmorItem;
import net.siegemc.core.items.Rarity;
import net.siegemc.core.items.Stat;
import org.bukkit.Material;
import org.bukkit.inventory.ItemStack;

import java.util.Arrays;

public class TestHelmet extends ArmorItem {
    public TestHelmet() {
        super("Test Helmet", new ItemStack(Material.LEATHER_HELMET, 1));
        this.setDescription(Arrays.asList("A quick item to test if armor", "works correctly"));
        this.setLevelRequirement(5);
        this.setRarity(Rarity.DEBUG);
        this.getStats().put(Stat.HEALTH, 30);
        Core.getItems().put("Test Helmet", this);
    }
}

File: src/main/kotlin/net/siegemc/core/items/implemented/armor/TestHelmet.kt
package net.siegemc.core.items.implemented.armor

import net.siegemc.core.items.CustomItemUtils
import net.siegemc.core.items.enums.Rarity
import net.siegemc.core.items.recipes.recipes
import net.siegemc.core.items.types.armor.CustomHelmet
import org.bukkit.Material
import org.bukkit.inventory.ItemStack

class TestHelmet() : CustomHelmet(
    name = "Test Helmet",
    customModelData = 1,
    description = listOf("A helmet for testing"),
    levelRequirement = 0,
    material = Material.DIAMOND_HELMET,
    recipeList = recipes {

    },
    baseStats = CustomItemUtils.statMap(strength = 10.0)
) {

    constructor(quality: Int): this() {
        this.quality = quality
        this.rarity = Rarity.getFromInt(quality)
    }

    constructor(item: ItemStack): this() {
        this.item = item
        deserialize()
    }

}